molgenis-app-gavin2/src/main/java/org/molgenis/app/gavin/GavinService.java

```diff
@@ -0,0 +1,17 @@
+package org.molgenis.app.gavin;
+
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import org.molgenis.app.gavin.meta.GavinRun;
+
+public interface GavinService {
+  String upload(HttpServletRequest httpServletRequest) throws IOException;
```

Comment:
Document the request requirements (multi-part form with one file? multi-part form with multiple files? multi-part form with one file and other inputs? binary data?)